pkg/infinity/client.go

```diff
@@ -202,26 +202,28 @@ func (client *Client) req(ctx context.Context, url string, body io.Reader, setti
 		logger.Error("url is not in the allowed list. make sure to match the base URL with the settings", "url", req.URL.String())
 		return nil, http.StatusUnauthorized, 0, errorsource.DownstreamError(errors.New("requested URL is not allowed. To allow this URL, update the datasource config Security -> Allowed Hosts section"), false)
 	}
-	logger.Debug("yesoreyeram-infinity-datasource plugin is now requesting URL", "url", req.URL.String())
+	logger.Debug("yesoreyeram-infinity-datasource plugin requesting URL", "host", req.URL.Hostname(), "url_path", req.URL.Path, "method", req.Method, "type", query.Type)
 	res, err := client.HttpClient.Do(req)
 	duration = time.Since(startTime)
+	logger.Debug("yesoreyeram-infinity-datasource plugin received response", "host", req.URL.Hostname(), "url_path", req.URL.Path, "method", req.Method, "type", query.Type, "duration_ms", duration.Milliseconds())
 	if res != nil {
 		defer res.Body.Close()
 	}
-	if err != nil && res != nil {
-		logger.Error("error getting response from server", "url", url, "method", req.Method, "error", err.Error(), "status code", res.StatusCode)
-		return nil, res.StatusCode, duration, errorsource.SourceError(backend.ErrorSourceFromHTTPStatus(res.StatusCode), fmt.Errorf("error getting response from %s", url), false)
-	}
-	if err != nil && res == nil {
+	if err != nil {
+		if res != nil {
+			logger.Error("error getting response from server", "url", url, "method", req.Method, "error", err.Error(), "status code", res.StatusCode)
+			return nil, res.StatusCode, duration, errorsource.SourceError(backend.ErrorSourceDownstream, fmt.Errorf("error getting response from %s", url), false)
+		}
 		logger.Error("error getting response from server. no response received", "url", url, "error", err.Error())
 		return nil, http.StatusInternalServerError, duration, errorsource.DownstreamError(fmt.Errorf("error getting response from url %s. no response received. Error: %w", url, err), false)
 	}
-	if err == nil && res == nil {
+	if res == nil {
 		logger.Error("invalid response from server and also no error", "url", url, "method", req.Method)
 		return nil, http.StatusInternalServerError, duration, errorsource.DownstreamError(fmt.Errorf("invalid response received for the URL %s", url), false)
 	}
 	if res.StatusCode >= http.StatusBadRequest {
-		return nil, res.StatusCode, duration, errorsource.SourceError(backend.ErrorSourceFromHTTPStatus(res.StatusCode), errors.New(res.Status), false)
+		err = fmt.Errorf("%w. %s", ErrUnsuccessfulHTTPResponseStatus, res.Status)
+		return nil, res.StatusCode, duration, errorsource.DownstreamError(err, false)
```

Comment:
```suggestion
		// Infinity can query anything and users are responsible for ensuring that endpoint/auth is correct
		// therefore any incoming error is considered downstream
		return nil, res.StatusCode, duration, errorsource.DownstreamError(err, false)
```
same as above, I'd add comment to explain why is the error always downstream. Not sure if my phrasing here is the best - feel free to update.